Changes

```diff
@@ -37,14 +37,17 @@ Working version
   (Stephen Dolan, Xavier Leroy and David Allsopp,
    review by Xavier Leroy and Gabriel Scherer)
 
-* #8711: The major GC hooks are no longer allowed to interact with the
+- #8711: The major GC hooks are no longer allowed to interact with the
    OCaml heap.
    (Jacques-Henri Jourdan, review by Damien Doligez)
 
 - #8691: Allocation functions are now guaranteed not to trigger any
    OCaml callback when called from C.
    (Jacques-Henri Jourdan, review by Stephen Dolan and Gabriel Scherer)
 
+- #8713: Introduce domain state to hold C global variables in the runtime.
```

Comment:
When written this way, I can never tell whether it means "to hold all C global variables" or "to hold some C global variables". Maybe that's because English is not my native language?

Reply:
You're not the only one to be confused (mandatory reference : https://en.wikipedia.org/wiki/United_Nations_Security_Council_Resolution_242#Interpretations )

Reply:
How about
```
* #8713: Introduce a state table in the runtime to contain the global variables
  which must be duplicated for each domain in the multicore runtime. For native
  code, this domain state is accessible via the exception pointer register (the
  exception pointer being part of the new domain state).
```
It's probably better to err on the side of marking this a breaking change (although the globals affected are not normally used outside the runtime, I think?). There are a couple of other reviewers who should be in the list too!

Reply:
English isn't my first language either. How about "Introduce domain state to hold all the domain-local C global variables in the runtime"?

Reply:
> For native code, this domain state is accessible via the exception pointer register

This is only true on amd64. On amd64, the domain state pointer is held in the register which used to hold the exception pointer. On arm, the domain state pointer is held in the register which used to hold the allocation limit; by stealing allocation limit, there is one additional load instruction during allocation. On power, s390x, arm64, there are enough general purpose registers available and hence we take over a general purpose register for domain state pointer. On i386, which has only a few registers, the domain state pointer is maintained in a global. 

Hence, I'd be happy with:

```
* #8713: Introduce a state table in the runtime to contain the global variables which 
must be duplicated for each domain in the multicore runtime.
```

Reply:
I'm not sure about marking it as a breaking change. We only do that for changes that have a chance of breaking reasonably-written programs, otherwise we mark everything, and the mark becomes useless.

Reply:
Agreed. I'd not consider this to be a breaking change if `compatibility.h` changes are reverted.

Reply:
`compatibility.h` changes have been reverted. There are not breaking changes now.